kagefunc.py

```diff
@@ -85,12 +85,12 @@ def generate_keyframes(clip: vs.VideoNode, out_path=None) -> None:
     # Speed up the analysis by resizing first. Converting to 8 bit also seems to improve the accuracy of wwxd.
     clip = core.resize.Bilinear(clip, 640, 360, format=vs.YUV420P8)
     clip = core.wwxd.WWXD(clip)
-    out_txt = "# WWXD log file, using qpfile format\n\n"
+    out_txt = "# WWXD log file, using qpfile format\n\n" if header else ""
     for i in range(clip.num_frames):
         if clip.get_frame(i).props.Scenechange == 1:
-            out_txt += "%d I -1\n" % i
-        if i % 1000 == 0:
-            print(i)
+            out_txt += f"{i} I -1\n" if i != 0 else ""
+        if i % 1 == 0:
```

Comment:
This just prints the status for every line, so the condition is always true, but why do you need that much output anyway? You shouldn’t have much filtering before this, so steps of 1000 frames shouldn’t take longer than a few seconds.

Reply:
I personally prefer being able to see that it's continuously busy. Especially since occasionally stuff randomly freezes for me, and having it only print a message every 1000 frames means I won't know when it's frozen until I notice it's taking forever until it pushes another update. This way I can see that it's actually not moving instead of guessing if it's still busy.

Reply:
make it a user defined variable. best of both worlds.